ivy/functional/frontends/torch/nn/functional/pooling_functions.py

```diff
@@ -62,16 +75,9 @@ def avg_pool2d(
     count_include_pad=True,
     divisor_override=None,
 ):
-    # Figure out input dims N
-    input_rank = input.ndim
-
-    if input_rank == 3:
-        # CHW
-        data_format = "CHW"
-    elif input_rank == 4:
-        # NCHW
-        data_format = "NCHW"
-
+    if stride is None:
+        stride = kernel_size
+    data_format = "NCHW"
     kernel_size = _broadcast_pooling_helper(kernel_size, "2d", name="kernel_size")
     stride = _broadcast_pooling_helper(stride, "2d", name="stride")
     padding = _broadcast_pooling_helper(padding, "2d", name="padding")
```

Comment:
As I commented in the other PR, please check whether we really need to use `_broadcast_pooling_helper` in all these functions.

Reply:
yeah, indeed it's handled internally in ivy.avg_pool1d but we need to broadcast these attrs to determine which padding algorithm should we use.

Reply:
Right, I just noticed that the avg pooling ivy functions only support `str` padding. I will make a separate task to extend this. For now as you said we have to keep the broadcasting of the padding here. 
I think however that the broadcasting of `kernel_size` is not needed.